src/hv_exc.c

```diff
@@ -91,12 +92,13 @@ static void _hv_exc_proxy(struct exc_info *ctx, uartproxy_boot_reason_t reason,
     }
 }
 
-static void hv_maybe_switch_cpu(struct exc_info *ctx, uartproxy_boot_reason_t reason, u32 type,
-                                void *extra)
+static void hv_rendezvous_cpu(struct exc_info *ctx, uartproxy_boot_reason_t reason, u32 type,
+                              void *extra)
 {
-    while (hv_want_cpu != -1) {
+    while (hv_want_cpu_in_rendezvous && hv_want_cpu != -1) {
```

Comment:
If there's a separate pinned CPU flag, then this function doesn't have to change at all.